src/build.d

```diff
@@ -721,6 +712,19 @@ void processEnvironment()
 /// Returns: all source files for the compiler
 auto sourceFiles()
 {
+    bool includeInBackend(string path)
+    {
+        auto filename = path.baseName;
+
+        auto result = filename != "dt.d" && filename != "obj.d";
+        version(Posix)
+            result = result && filename != "machobj.d" && filename != "mscoffobj.d";
+        else version(Windows)
+            result = result && filename != "elfobj.d" && filename != "machobj.d";
+
+        return result;
+    }
```

Comment:
The exclusion was just a temporary workaround. We can do this with version declarations at the top of these platform specific files - we have done this before for some of them and it allows us to get rid of this logic.

Reply:
Example: https://github.com/dlang/dmd/blob/master/src/dmd/libmach.d

Reply:
I see now.  It appears that has already been done.  I've updated build.d and it seems to work.  Thanks.